### DISCUSSION TITLE
APIs

### DISCUSSION THREAD

**Discussion by tfbecker** [twentyhq/twenty#1701] [General]
I would love to programmatically populate Twenty (CRUD).
Is this on the roadmap?

**Comment by Bonapara** [comment]
Hi @tfbecker, thanks for reaching out! We are currently working on this and it is part of our roadmap. We anticipate having an MVP ready in 2 weeks, followed by a stable version in 2 months. (This release will also feature custom objects)

What would you like to build with the API?

**Reply by tfbecker** [reply]
Hi Bonapara, 
I will use the APIs to automatically add new leads that I scrape somewhere else + data enrichment.

**Reply by Bonapara** [reply]
Thank you for detailing your use case. API development began on Monday!

**Comment by FelixMalfait** [comment]
Closing as GraphQL API and Rest API have been shipped! (You can generate a token in Settings > Developers)
Rest API is not documented yet, we'll need to figure out a way to generate it dynamically (started here https://github.com/twentyhq/twenty/pull/2900)
